build/ci.go

```diff
@@ -264,10 +265,53 @@ func doTest(cmdline []string) {
 	if *parallel != 0 {
 		gotest.Args = append(gotest.Args, "-p", strconv.Itoa(*parallel))
 	}
-	if *coverage {
-		gotest.Args = append(gotest.Args, "-covermode=atomic", "-cover", "-coverprofile=coverage.out")
+	gotest.Args = append(gotest.Args, packages...)
+	build.MustRun(gotest)
+}
+
+func doCover(cmdline []string) {
+	var (
+		parallel   = flag.Int("p", 0, "The number of parallel coverage test executions (default: the number of CPUs available)")
+		excludes   = flag.String("exclude", "", "Comma-separated top-level directories to be excluded in coverage test")
+		outputFile = flag.String("coverprofile", "coverage.out", "The coverage profile file will be generated by coverage test")
+	)
+	flag.CommandLine.Parse(cmdline)
+	env := build.Env()
+
+	packages := []string{"./..."}
+	if len(flag.CommandLine.Args()) > 0 {
+		packages = flag.CommandLine.Args()
+	}
+	packages = build.ExpandPackagesNoVendor(packages)
+	if *excludes != "" {
+		packages = build.ExcludePackages(packages, strings.Split(*excludes, ","))
+	}
+
+	coverPackages := []string{"./..."}
+	coverExcludes := []string{
+		"/tests",
+		"/metric",
+		"/build",
+		"/client",
```

Comment:
How about adding `/contracts`? PR #113 will move a test file to node/sc directory.

Reply:
Thank you for the recommendation. We are considering the option also. The excluding list will be discussed in other PR later.